tests/integration/golang/helpers/context.go

```diff
@@ -0,0 +1,24 @@
+package helpers
+
+import (
+	"encoding/json"
+	"strings"
+)
+
+// ExtractContextBytes finds the metric context in decoded data, using the given key prefix,
+// and marshals to bytes.
+func ExtractContextBytes(contextPrefix string, decodedData map[string]any) ([]byte, error) {
+	contx := ExtractContext(contextPrefix, decodedData)
+	return json.Marshal(contx)
+}
+
+// ExtractContextBytes finds the metric context in the decoded data, using the given key prefix.
+func ExtractContext(contextPrefix string, decodedData map[string]any) map[string]any {
+	contx := map[string]any{}
+	for key := range decodedData {
+		if strings.HasPrefix(key, contextPrefix) && len(key) > len(contextPrefix) {
+			contx[key[len(contextPrefix)+1:]] = decodedData[key]
+		}
+	}
+	return contx
+}
```

Comment:
Im not sure that this is what we exactly need. so:
- extraction should be a part of our `decoder`.
- extraction should work for any object and not only for context out of the box.
- `json` object could be complex object with other nested objects and extraction should work for this case as well under the hood of `decoder`.

Reply:
saving this improvement for another day -- possibly conversion to arrow records will happen first.